### PR TITLE
Create Volumetric Task

### PR DESCRIPTION
**Pull request by kieftrav**
Staging branch URL: https://pr-7227.pfe-preview.zooniverse.org

Describe your changes.
- Rename experimental flag for the Volumetric Viewer to "volumetricProject" from "volumetricViewer"
- Create Volumetric task that it can be added to any project with the "volumetricProject" and "FemLab" flags enabled

# Required Manual Testing
- Open the [Admin Page](https://local.zooniverse.org:3735/admin/project_status/kieftrav/volumetric-viewer) and the [Workflow Editor](https://local.zooniverse.org:3735/lab/2015/workflows/3831) in separate tabs for my test/staging Volumetric Viewer project [Volumetric Viewer]() project for testing
- As of this writing, "FemLab" and "volumetricViewer" are enabled and you should see a Volumetric task existing on workflow 3831

### Things you should be able to do:
- [ ] Removing the Volumetric Task
- [ ] Adding a new Volumetric Task
- [ ] Saving / refreshing and the Volumetric Task still exists + works
- [ ] Disabling the "volumetricViewer" flag should prevent the use from adding a Volumetric task
- [ ] You should be able to add the Volumetric to any other project of your choosing as long as the two experimental flags are enabled

### Requests for PR Review
- The Volumetric task was _heavily_ inspired by the Highlighter task. I do not know if everything in that file is necessary/right/appropriate/correct. This is the file that might need the most review/iteration because I lack context on all its implications elsewhere in the codebase
- If there's something glaringly obvious/missing, please schedule a call/screenshare to walk through the details (I prefer that to a voluminous wall of text).
- If there is a place to write a test, I'm happy to do that. I'm not sure what/where would make sense though, if so.

### Standard Tests
- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist
- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `npm ci` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on GitHub Actions?

# Optional
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Affects: Project Builder -> Workflow Editor (FEM Lab only)

This PR adds the **"VolumetricTask"** Task to the Workflow Editor.

On the project builder:
- If a project has the `volumetricViewer` and `FEM Lab` experimental tools enabled, then the "Volumetric Task" will be listed as an option when adding a new Task.

On the PFE classifier:
- If a volunteer attempts to load a Volumetric Task on the PFE classifier, then the Task area will just show a placeholder with no input controls.
- The Volumetric Task is only valid on the FEM classifier.

Functionality checks out fine, but I've added a few comments to improve the code.

### Testing

Testing project builder, on localhost:
- Enable required experimental tools on a test project.
- Go to, or create, a test workflow, e.g. https://local.zooniverse.org:3735/lab/1982/workflows/3841?femLab=true
- Add a Volumetric Task.
- Add Instructions and Help text to the task.
- Save, and then reload page to confirm changes.

I'm able to create and edit a VolumetricTask with no issues.

⚠️  NOTE: whether or not the structure of the VolumetricTask is correct for the needs of the corresponding code on FEM is a separate question.

Testing PFE classifier, on localhost:
- View the workflow on the PFE classifier: https://local.zooniverse.org:3735/projects/darkeshard/example-1982/classify?reload=0&workflow=3841

Workflow loads fine. A placeholder is show in place of the Task. Nothing crashes.

Testing FEM classifier, on localhost, with app-project:
- Boot up FEM's app-project, on whatever branch of master is current.
- View the workflow on the FEM classifier: https://local.zooniverse.org:3000/projects/darkeshard/example-1982/classify/workflow/3841?env=staging

Workflow loads fine. 

⚠️ NOTE: a more thorough testing of the interaction between VolumetricTask on PFE project builder + Volumetric Task on FEM Classifier should be done with a project/workflow that's FULLY set up to use the Volumetric Viewer + Volumetric Viewer Data + Volumetric Task (e.g. https://local.zooniverse.org:3000/projects/kieftrav/volumetric-viewer/classify/workflow/3831?env=staging) but that's a step too far for this PR review.

### Status

LGTM, this meets minimum safety checks - functionality works as advertised for the project builder, and nothing seems to crash on PFE as a whole. 👍 

I understand that the VolumetricTask is still in development, so my questions about e.g. "what's the structure of the VolumetricTask _annotation?"_ should be deferred until those details are confirmed. (Or maybe I missed the PR/ADR on FEM.) We can always revisit this code to tweak, anyhoo.

Regarding my comments on the code in this PR:
- my main concern is that the default props, prop types, etc shouldn't add code that's unused, e.g. I don't think VolumetricTask uses tools.
  - I'm assuming things like tools, _toolIndex etc are leftovers from using the code pattern from similar Tasks (Drawing Task?)
  - If I'm wrong about this, please let me know - perhaps I need to those questions about annotation structure earlier.
- I'm also in favour of making the code clearer for future devs to read, so some things like the placeholder text on render() could use a minor text improvement.

File: app/classifier/tasks/volumetric/index.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+VolumetricTask.getTaskText = (task) => task.instruction;
+
+VolumetricTask.getDefaultAnnotation = () => ({ _toolIndex: 0, value: [] });
+
+VolumetricTask.defaultProps = {
+  showRequiredNotice: false,
+  task: {
+    help: "",
+    type: "volumetric",
+    instruction: "Describe how to use this tool",
```
[the remark this answers is not in the document]
Comment (kieftrav) [reply app/classifier/tasks/volumetric/index.jsx]: I don't like implicit things in code like this... So I've made it explicit

File: app/classifier/tasks/volumetric/index.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  workflow: {
+    tasks: [],
+  },
```
[the remark this answers is not in the document]
Comment (kieftrav) [reply app/classifier/tasks/volumetric/index.jsx:33]: Awesome - thank you for the context. Removed.

File: app/classifier/tasks/volumetric/index.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    help: PropTypes.string,
+    instruction: PropTypes.string,
+    required: PropTypes.bool,
+    tools: PropTypes.array,
+    type: PropTypes.string,
+  }),
+  workflow: PropTypes.shape({
+    tasks: PropTypes.object,
+  }),
+};
```
[the remark this answers is not in the document]
Comment (kieftrav) [reply app/classifier/tasks/volumetric/index.jsx]: Removed - didn't need them.

File: app/classifier/tasks/volumetric/index.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import PropTypes from "prop-types";
+import React from "react";
+import GenericTaskEditor from "../generic-editor";
+
+export default class VolumetricTask extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    return <div>{"VolumetricTask Tool"}</div>;
```
[the remark this answers is not in the document]
Comment (kieftrav) [reply app/classifier/tasks/volumetric/index.jsx]: Agreed and implemented.

**Pull request merged**